common/src/com/intellij/plugins/haxe/config/HaxeTarget.java

```diff
@@ -107,4 +108,25 @@ public static void initCombo(@NotNull DefaultComboBoxModel comboBoxModel) {
   public String toString() {
     return description;
   }
+
+  /**
+   * Match the string against the compiler's argument/parameter/flag for the
+   * target output.
+   *
+   * @param compilerTargetArgument - string to compare, e.g. '-js', '-neko'
+   * @return The target matching the flag, or null, if not found.
+   */
+  @Nullable
+  public static HaxeTarget matchOutputTarget(String compilerTargetArgument) {
+    for (HaxeTarget t : HaxeTarget.values()) {
+      if (t.getCompilerFlag().equals(compilerTargetArgument)) {
+        return t;
+      }
+    }
+    // as3 is an old case.
+    if ("-as3".equals(compilerTargetArgument)) {
```

Comment:
OK.  I'll change it.  But first, I'll check the lime tool to see what it does with -as3.